pkg/daemon/ceph/client/pool.go

```diff
@@ -471,7 +471,7 @@ func createReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 
 	logger.Infof("reconciling replicated pool %s succeeded", pool.Name)
 
-	if checkFailureDomain {
+	if checkFailureDomain || pool.PoolSpec.DeviceClass != "" {
```

Comment:
we don't need to check similar to https://github.com/rook/rook/blob/72452f0558281c0b57f93017c481f5518ca12b86/pkg/daemon/ceph/client/pool.go#L483-L486
since deviceClass can be empty or not always specified

Reply:
If failureDomain is blank, but the deviceClass is changed, how do we get past the return on line 485?

Reply:
I missed that since in the pool example failure domain was set, now I have added to check for deviceClass also.